src/PowerSystems2PRAS.jl

```diff
@@ -229,160 +115,180 @@ function generate_pras_system(
-    start_id = Array{Int64}(undef, num_regions)
-    region_gen_idxs = Array{UnitRange{Int64}, 1}(undef, num_regions)
-    reg_wind_gens = []
-    reg_pv_gens = []
+# Generator must not have HydroEenergyReservoir or have 0 max active power or be a hybrid system
+"""
+    $(TYPEDSIGNATURES)
 
-    if (lump_region_renewable_gens)
-        for (idx, region) in enumerate(regions)
-            reg_ren_comps = get_available_components_in_aggregation_topology(
-                PSY.RenewableGen,
-                sys,
-                region,
-            )
-            wind_gs = filter(
-                x -> (PSY.get_prime_mover_type(x) == PSY.PrimeMovers.WT),
-                reg_ren_comps,
-            )
-            pv_gs = filter(
-                x -> (PSY.get_prime_mover_type(x) == PSY.PrimeMovers.PVe),
-                reg_ren_comps,
-            )
-            reg_gen_comps =
-                get_available_components_in_aggregation_topology(PSY.Generator, sys, region)
-            gs = filter(
-                x -> (
-                    ~(typeof(x) == PSY.HydroEnergyReservoir) &&
-                    ~(iszero(PSY.get_max_active_power(x))) &&
-                    PSY.IS.get_uuid(x) ∉ union(
-                        s2p_meta.hs_uuids,
-                        PSY.IS.get_uuid.(wind_gs),
-                        PSY.IS.get_uuid.(pv_gs),
-                    )
-                ),
-                reg_gen_comps,
-            )
-            push!(gens, gs)
-            push!(reg_wind_gens, wind_gs)
-            push!(reg_pv_gens, pv_gs)
+Extraction of generators using formulation dictionary to create a list of generators
+and appropriate indices for PRAS. Note that objects with 0 max active power are excluded.
+"""
+function get_generator_region_indices(
+    sys::PSY.System,
+    s2p_meta::S2P_metadata,
+    regions,
+    component_to_formulation::Dict{PSY.Device, GeneratorFormulation},
+)
+    gens = Array{PSY.Device}[]
+    start_id = Array{Int64}(undef, length(regions))
+    region_gen_idxs = Array{UnitRange{Int64}, 1}(undef, length(regions))
 
-            if (idx == 1)
-                start_id[idx] = 1
-            else
-                if (length(reg_wind_gens[idx - 1]) > 0 && length(reg_pv_gens[idx - 1]) > 0)
-                    start_id[idx] = start_id[idx - 1] + length(gens[idx - 1]) + 2
-                elseif (
-                    length(reg_wind_gens[idx - 1]) > 0 || length(reg_pv_gens[idx - 1]) > 0
-                )
-                    start_id[idx] = start_id[idx - 1] + length(gens[idx - 1]) + 1
-                else
-                    start_id[idx] = start_id[idx - 1] + length(gens[idx - 1])
-                end
-            end
+    reg_wind_gens = []
+    reg_pv_gens = []
+    for (idx, region) in enumerate(regions)
+        reg_ren_comps = filter(
+            x -> haskey(component_to_formulation, x),
+            get_available_components_in_aggregation_topology(PSY.RenewableGen, sys, region),
```

Comment:
Do we need to do this again _get_available_components_in_aggregation_topology_ ? Or can we get all components from the Dict with this formulation and filter for this?

Reply:
I'm not sure we have a function to group by aggregation topology from a dictionary yet. It might be appropriate, but I didn't want to get into that here.

Reply:
https://github.com/NREL-Sienna/SiennaPRASInterface.jl/issues/62